File: app/src/main/java/br/com/vicentec12/mygames/data/source/local/AppDatabase.java
package br.com.vicentec12.mygames.data.source.local;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import br.com.vicentec12.mygames.data.model.Game;
import br.com.vicentec12.mygames.data.source.local.dao.GameDao;
import br.com.vicentec12.mygames.util.AppExecutors;

@Database(entities = {Game.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    private static final String LOG_TAG = AppDatabase.class.getSimpleName();
    private static final Object LOCK = new Object();
    private static final String DATABASE_NAME = "mygames.db";
    private static AppDatabase sInstance;

    public static AppDatabase getInstance(Context context) {
        if (sInstance == null) {
            synchronized (LOCK) {
                sInstance = Room.databaseBuilder(context.getApplicationContext(),
                        AppDatabase.class, AppDatabase.DATABASE_NAME).build();
            }
        }
        return sInstance;
    }

    public abstract GameDao getGameDao();

    /**
     * Método responsável por limpar todas as tabelas do banco de dados interno.
     */
    public static void clearDatabase(Context context) {
        AppExecutors.getInstance().diskIO().execute(() -> {
            getInstance(context).getGameDao().deleteAll();
        });
    }
}

File: app/src/main/java/br/com/vicentec12/mygames/data/local/AppDatabase.kt
package br.com.vicentec12.mygames.data.local

import androidx.room.Database
import androidx.room.RoomDatabase
import br.com.vicentec12.mygames.data.local.entities.ConsoleEntity
import br.com.vicentec12.mygames.data.local.entities.GameEntity
import br.com.vicentec12.mygames.data.local.dao.ConsoleDao
import br.com.vicentec12.mygames.data.local.dao.GameDao

@Database(entities = [ConsoleEntity::class, GameEntity::class], version = 1)
abstract class AppDatabase : RoomDatabase() {

    abstract fun getConsoleDao(): ConsoleDao

    abstract fun getGameDao(): GameDao

    companion object {

        const val DATABASE_NAME = "mygames.db"

    }

}